src/auth/SchemeConfig.cc

```diff
@@ -113,20 +111,15 @@ Auth::SchemeConfig::parse(Auth::SchemeConfig * scheme, int, char *param_str)
     } else if (strcmp(param_str, "key_extras") == 0) {
         keyExtrasLine = ConfigParser::NextQuotedToken();
         Format::Format *nlf =  new ::Format::Format(scheme->type());
-        if (!nlf->parse(keyExtrasLine.termedBuf())) {
-            debugs(29, DBG_CRITICAL, "FATAL: Failed parsing key_extras formatting value");
-            self_destruct();
-            return;
-        }
+        nlf->parse(keyExtrasLine.termedBuf()); // XXX: throws and leaks nlf on errors
```

Comment:
Cant' we catch and rethrow? Or is this left as a followup?

Reply:
I do not see any use for a re-throw from here. What are you thinking we do at this point in the code?

Note that the whole line being parsed is displayed elsewhere, so we do not need to embellish the thrown error with much context. Just indicate clearly the exact part which went wrong within that line.

Reply:
Got you. Good.